### ISSUE TITLE
liq_cue_file comes back numeric instead of original value: News cut off, autocue sometimes returns no result

### ISSUE THREAD

**Issue by Moonbase59** [AzuraCast/AzuraCast#7310]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #ba75595 (2024-07-09 4:02)

### What happened?

I suspect some issue in the new "store and overwrite everything" tag logic in AzuraCast. I have this data in the file:

![Auswahl_471](https://github.com/AzuraCast/AzuraCast/assets/3706922/8130ab66-54b9-4f36-8fa0-a32cb546d018)

and it comes back as:

```
2024/07/10 07:00:23 [show_meta:3] ("duration", "145.54")
2024/07/10 07:00:23 [show_meta:3] ("liq_amplify", "-6.21 dB")
2024/07/10 07:00:23 [show_meta:3] ("liq_amplify_adjustment", "0.00 dB")
2024/07/10 07:00:23 [show_meta:3] ("liq_autocue", "cue_file")
2024/07/10 07:00:23 [show_meta:3] ("liq_blank_skipped", "false")
2024/07/10 07:00:23 [show_meta:3] ("liq_blankskip", "0.00")
2024/07/10 07:00:23 [show_meta:3] ("liq_cross_duration", "2.50")
2024/07/10 07:00:23 [show_meta:3] ("liq_cross_start_next", "142.50")
2024/07/10 07:00:23 [show_meta:3] ("liq_cue_duration", "144.90")
2024/07/10 07:00:23 [show_meta:3] ("liq_cue_file", "1.00")
2024/07/10 07:00:23 [show_meta:3] ("liq_cue_in", "0.00")
2024/07/10 07:00:23 [show_meta:3] ("liq_cue_out", "144.90")
2024/07/10 07:00:23 [show_meta:3] ("liq_fade_in", "0.10")
2024/07/10 07:00:23 [show_meta:3] ("liq_fade_in_delay", "2.40")
2024/07/10 07:00:23 [show_meta:3] ("liq_fade_out", "2.40")
2024/07/10 07:00:23 [show_meta:3] ("liq_fade_out_delay", "0.10")
2024/07/10 07:00:23 [show_meta:3] ("liq_longtail", "false")
2024/07/10 07:00:23 [show_meta:3] ("liq_loudness", "-11.79 LUFS")
2024/07/10 07:00:23 [show_meta:3] ("liq_loudness_range", "3.82 LU")
2024/07/10 07:00:23 [show_meta:3] ("liq_reference_loudness", "-18.00 LUFS")
2024/07/10 07:00:23 [show_meta:3] ("liq_sustained_ending", "true")
2024/07/10 07:00:23 [show_meta:3] ("liq_true_peak", "1.25")
2024/07/10 07:00:23 [show_meta:3] ("liq_true_peak_db", "1.90 dBFS")
2024/07/10 07:00:23 [show_meta:3] ("media_id", "323203")
2024/07/10 07:00:23 [show_meta:3] ("replaygain_reference_loudness", "-18.00 LUFS")
2024/07/10 07:00:23 [show_meta:3] ("replaygain_track_gain", "-6.21 dB")
2024/07/10 07:00:23 [show_meta:3] Now playing: Nite Radio - Nachrichten
```

so clearly overwriting the data from the actual news MP3 with the data from an hour back. This _cuts the news short by almost 20s_ in this case and apparently also sometimes leads to _no Autocue results coming back at all_ (and getting loudness jumps)  because `liq_cue_file` gets checked for "not there" or "true" later in the code (after it’s been checked for "false").

I have a mechanism in place to fight the AzuraCast override with an "override the override" but that can only function if `liq_cue_file` isn’t converted to "1.00" as shown above. **Suspect something in the code makes it a "truthy" numeric.**

See also:
- https://github.com/Moonbase59/autocue/issues/33
- https://github.com/Moonbase59/autocue/blob/master/FAQ.md#what-exactly-is-liq_cue_file-for-and-should-i-use-it

---

**Explanation for users:**

The original idea behind AzuraCast storing the data in its database was to be able to provide the tag data to Autocue to speed up operations in case we have a remote or read-only filesystem, or cannot pre-tag. AzuraCast picks up the last calculation result and saves it for re-use.

---

**Temporary workaround for users needing current data (like news):**

You _could_ set
```
settings.autocue.cue_file.use_json_metadata := false
```
which would _completely_ ignore what AzuraCast tries to override, but then you also cannot use the _cue-point_, _start-next_ and _fade_ overrides you specify in the Visual Cue Editor/Advanced tab.

AzuraCast overrides are applied _after_ any Autocue calculations, effectively overwriting anything `cue_file` has calculated. That’s because AzuraCast uses _annotations_, which rank top priority in the priority scheme _Defaults → File Tags → Annotations_, see https://moonbase59.github.io/autocue/presentation/autocue.html#/metadata.

**Comment by BusterNeece** [comment]
@Moonbase59 Should be better now.

**Comment by Moonbase59** [comment]
Thanks. I’m on Rolling Release #[536b5f6](https://github.com/AzuraCast/AzuraCast/commit/536b5f677304cedd497c4d35c67c2ccc8b23f1b0) (2024-07-10 8:05) now, and will watch it for a few days, since the other bug only happened 8 times in 5 days here.

First time round, it presented `liq_cue_file=1.00` again as expected (from the DB). Let’s see if it now corrects.

---

For all that used the **temporary measure above: Don’t forget to reset your JSON setting again _after_ upgrading to Rolling Release #[536b5f6](https://github.com/AzuraCast/AzuraCast/commit/536b5f677304cedd497c4d35c67c2ccc8b23f1b0) or newer!** You can simply comment it out (the default is `true`), or explicitly set it to `true` again:

```
settings.autocue.cue_file.use_json_metadata := true
```

**Comment by Moonbase59** [comment]
Hm. At 16:00 I still got
```
2024/07/10 16:00:23 [show_meta:3] ("liq_cue_file", "1.00")
```

For good measure, I also checked to 16:00 news file directly on the station again:
```
$ ffprobe -hide_banner news-de.mp3 
Input #0, mp3, from 'news-de.mp3':
  Metadata:
    title           : Nachrichten
    artist          : Nite Radio
    genre           : News
    liq_cue_file    : true
    liq_reference_loudness: -18.00 LUFS
    liq_cue_duration: 159.30
    liq_cue_in      : 0.00
    liq_cue_out     : 159.30
    liq_cross_start_next: 156.90
    liq_longtail    : false
    liq_sustained_ending: true
    liq_loudness    : -13.01 LUFS
    liq_loudness_range: 4.57 LU
    liq_amplify     : -4.99 dB
    liq_amplify_adjustment: 0.00 dB
    liq_blankskip   : 0.00
    liq_blank_skipped: false
    liq_true_peak   : 1.299000
    liq_true_peak_db: 2.27 dBFS
    replaygain_track_gain: -4.99 dB
    replaygain_track_peak: 1.299000
    replaygain_track_range: 4.57 dB
    replaygain_reference_loudness: -18.00 LUFS
    encoder         : Lavf58.76.100
  Duration: 00:02:39.89, start: 0.011021, bitrate: 256 kb/s
  Stream #0:0: Audio: mp3, 48000 Hz, stereo, fltp, 256 kb/s
    Metadata:
      encoder         : Lavf
    Side data:
      replaygain: track gain - -4.990000, track peak - 0.000030, album gain - unknown, album peak - unknown, 
```

I now started a manual re-process in AzuraCast. Let’s see if that changes anything.

**Comment by Moonbase59** [comment]
No, doesn’t work. Maybe the media checker/reprocess doesn’t overwrite it from the file’s tags once the `1.00` is stored?

AzuraCast annotates `liq_cue_file=1.00`, gives it to Autocue which can’t handle that value, metadata gets returned as `1.00` in the feedback. Loop.

Here’s a complete level 4 log, from reloading the playlist at 16:59:45 until finishing the request at 17:03:01:
[AzuraCast 536b5f6 liq_cue_file.log](https://github.com/user-attachments/files/16164499/AzuraCast.536b5f6.liq_cue_file.log)

**Comment by LordHelmchen666** [comment]
I don't know if this is related, but i think it is. Azuracast stopped updating meta tags automatically. When i download (hourly) my news file and tag it with cue_file it stores correct tags to file, but it seems Azuracast won't recognize there are new tags. When i run _CheckMediaTask_ from System-Debugger the new values are not updated/stored. When i add the file manually in Mediamanager to reprocess and then run _CheckMediaTask_ it fills in the new values.
As workaround i send values via API which works well.

**Comment by BusterNeece** [comment]
@Moonbase59 Found some minor bugs that may have affected reprocessing, both automatic and manually marked reprocessing.

The system should definitely be overwriting that metadata from the file if the file mtime is newer than the mtime in the DB.

I can confirm that just storing our feedback cache data _doesn't_ update the mtime in the database, so it should still be following normal reprocessing rules.

**Comment by Moonbase59** [comment]
Thanks for looking into it. It seems once it’s wrong, you’ll never get rid of it anymore (still showing `liq_cue_file=1.00`), because

- new file comes in
- media checker has no chance to update before playout
- old DB data is annotated for playout
- feedback saves old (wrong) back to DB

So next time you _still_ get the saved `1.00`.

I now ssh’ed in to the server and manually _deleted_ the file.

Hoping this gets picked up and next time round AzuraCast will provide the "new" old value in its annotations…

**Comment by BusterNeece** [comment]
@Moonbase59 I imagine you might've missed it, but one of the things I added in tonight's commits is the ability to batch clear extra metadata from tracks, then trigger an immediate reload from disk. You can do that via either the `azuracast:media:clear-extra` command (same supported syntax as `azuracast:media:reprocess`) or via the "Clear Extra Metadata" option on the Media Manager (or the `clear-extra` batch action via the API).

**Comment by Moonbase59** [comment]
Did indeed miss that, will try on the next round (full hour).

**Comment by Moonbase59** [comment]
On fresh read after #7314 fixed, without news ever played (so no feedback yet):

```
azuracast@e46980ba8825:~/stations/niteradio/playlists$ cat playlist_nachrichten.m3u 
autocue:annotate:title="Nachrichten",artist="Nite Radio",duration="165.02",song_id="89f90c2380b5ff2053b95c999f32feba",media_id="323883",liq_amplify="-4.78 dB",liq_amplify_adjustment="0.00 dB",liq_blank_skipped="false",liq_blankskip="0.00",liq_cross_start_next="162.10",liq_cue_duration="164.50",liq_cue_file="1.00",liq_cue_in="0.00",liq_cue_out="164.50",liq_longtail="false",liq_loudness="-13.22 LUFS",liq_loudness_range="4.45 LU",liq_reference_loudness="-18.00 LUFS",liq_sustained_ending="true",liq_true_peak="1.22",liq_true_peak_db="1.76 dBFS",replaygain_reference_loudness="-18.00 LUFS",replaygain_track_gain="-4.78 dB",replaygain_track_peak="1.22",replaygain_track_range="4.45 dB",playlist_id="539":media:time/news-de.mp3azuracast@e46980ba8825:~/stations/niteradio/playlists$ 
```

Nte the `liq_cue_file="1.00"`.

**Comment by Moonbase59** [comment]
Maybe here? → https://github.com/AzuraCast/AzuraCast/blob/35a447bbc618500b72f6531fac144d445ffb2f50/backend/src/Radio/AutoDJ/Annotations.php#L147

**Comment by Moonbase59** [comment]
Added the PR code to mystations, and works fine here:

```
2024/07/11 13:59:45 [playlist_nachrichten:3] Reloading playlist with URI /var/azuracast/stations/niteradio/playlists/playlist_nachrichten.m3u.
2024/07/11 13:59:45 [decoder.ffmpeg:3] Requested content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}
2024/07/11 13:59:45 [decoder.ffmpeg:3] FFmpeg recognizes "/var/azuracast/stations/niteradio/media/time/news-de.mp3" as audio: {codec: mp3, 48000Hz, 2 channel(s)}
2024/07/11 13:59:45 [decoder.ffmpeg:3] Decoded content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}
2024/07/11 13:59:45 [autocue.cue_file:3] Now autocueing: "/var/azuracast/stations/niteradio/media/time/news-de.mp3"
2024/07/11 13:59:45 [autocue.cue_file:3] Blank (silence) skipping active: false, set to 0. s
2024/07/11 13:59:45 [autocue.cue_file:3] Clipping prevention active: true
2024/07/11 13:59:45 [autocue.cue_file:3] Writing tags: false, including ReplayGain: false
2024/07/11 13:59:45 [autocue.cue_file:3] Writing metadata to /tmp/cue_file9f63c2.json
2024/07/11 13:59:45 [autocue.cue_file:3] cue_file result for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {"duration": 156.864, "liq_cue_duration": 164.5, "liq_cue_in": 0.0, "liq_cue_out": 164.5, "liq_cross_start_next": 162.1, "liq_longtail": false, "liq_sustained_ending": true, "liq_loudness": "-13.22 LUFS", "liq_loudness_range": "4.45 LU", "liq_amplify": "-4.78 dB", "liq_amplify_adjustment": "0.00 dB", "liq_reference_loudness": "-18.00 LUFS", "liq_blankskip": 0.0, "liq_blank_skipped": false, "liq_true_peak": 1.22, "liq_true_peak_db": "1.76 dBFS"}
2024/07/11 13:59:45 [autocue.cue_file:3] cue_file results override existing metadata because liq_cue_file=true tells us to.
2024/07/11 13:59:45 [autocue.cue_file:3] Clipping prevention: Adjusted calculated replaygain_track_gain from -4.78 dB to -4.78 dB
2024/07/11 13:59:45 [autocue.cue_file:2] Given fade-out duration (2.4 s) exceeds available time, using 2.4 s.
2024/07/11 13:59:45 [autocue.cue_file:3] Metadata added/corrected for "/var/azuracast/stations/niteradio/media/time/news-de.mp3":
2024/07/11 13:59:45 [autocue.cue_file:3] ("duration", "156.864")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_amplify", "-4.78 dB")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_amplify_adjustment", "0.00 dB")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_autocue", "cue_file")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_blank_skipped", "false")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_blankskip", "0.")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cross_duration", "2.50")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cross_start_next", "162.1")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cue_duration", "164.50")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cue_file", "true")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cue_in", "0.")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_cue_out", "164.5")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_fade_in", "0.1")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_fade_in_delay", "2.50")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_fade_out", "2.4")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_fade_out_delay", "0.10")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_longtail", "false")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_loudness", "-13.22 LUFS")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_loudness_range", "4.45 LU")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_reference_loudness", "-18.00 LUFS")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_sustained_ending", "true")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_true_peak", "1.22")
2024/07/11 13:59:45 [autocue.cue_file:3] ("liq_true_peak_db", "1.76 dBFS")
2024/07/11 13:59:45 [autocue.cue_file:3] ("replaygain_reference_loudness", "-18.00 LUFS")
2024/07/11 13:59:45 [autocue.cue_file:3] ("replaygain_track_gain", "-4.78 dB")
2024/07/11 13:59:45 [decoder.ffmpeg:3] Requested content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}
2024/07/11 13:59:45 [decoder.ffmpeg:3] FFmpeg recognizes "/var/azuracast/stations/niteradio/media/time/news-de.mp3" as audio: {codec: mp3, 48000Hz, 2 channel(s)}
2024/07/11 13:59:45 [decoder.ffmpeg:3] Decoded content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}

...

2024/07/11 14:00:23 [playlist_nachrichten:3] Prepared "/var/azuracast/stations/niteradio/media/time/news-de.mp3" (RID 143).
2024/07/11 14:00:23 [switch.11:3] Switch to news_fallback with transition.
2024/07/11 14:00:23 [replay_metadata.100:3] Content type is {audio=pcm(stereo)}.
2024/07/11 14:00:23 [news_fallback:3] Switch to news_switch.
[mp3float @ 0x7efeae5f6000] Could not update timestamps for skipped samples.
2024/07/11 14:00:23 [replay_metadata.101:3] Content type is {audio=pcm(stereo)}.
2024/07/11 14:00:23 [news_switch:3] Switch to playlist_nachrichten.
2024/07/11 14:00:23 [replay_metadata.102:3] Content type is {audio=pcm(stereo)}.
2024/07/11 14:00:23 [show_meta:3] ("duration", "165.02")
2024/07/11 14:00:23 [show_meta:3] ("liq_amplify", "-4.78 dB")
2024/07/11 14:00:23 [show_meta:3] ("liq_amplify_adjustment", "0.00 dB")
2024/07/11 14:00:23 [show_meta:3] ("liq_autocue", "cue_file")
2024/07/11 14:00:23 [show_meta:3] ("liq_blank_skipped", "false")
2024/07/11 14:00:23 [show_meta:3] ("liq_blankskip", "0.00")
2024/07/11 14:00:23 [show_meta:3] ("liq_cross_duration", "2.50")
2024/07/11 14:00:23 [show_meta:3] ("liq_cross_start_next", "162.10")
2024/07/11 14:00:23 [show_meta:3] ("liq_cue_duration", "164.50")
2024/07/11 14:00:23 [show_meta:3] ("liq_cue_file", "true")
2024/07/11 14:00:23 [show_meta:3] ("liq_cue_in", "0.00")
2024/07/11 14:00:23 [show_meta:3] ("liq_cue_out", "164.50")
2024/07/11 14:00:23 [show_meta:3] ("liq_fade_in", "0.10")
2024/07/11 14:00:23 [show_meta:3] ("liq_fade_in_delay", "2.50")
2024/07/11 14:00:23 [show_meta:3] ("liq_fade_out", "2.40")
2024/07/11 14:00:23 [show_meta:3] ("liq_fade_out_delay", "0.10")
2024/07/11 14:00:23 [show_meta:3] ("liq_longtail", "false")
2024/07/11 14:00:23 [show_meta:3] ("liq_loudness", "-13.22 LUFS")
2024/07/11 14:00:23 [show_meta:3] ("liq_loudness_range", "4.45 LU")
2024/07/11 14:00:23 [show_meta:3] ("liq_reference_loudness", "-18.00 LUFS")
2024/07/11 14:00:23 [show_meta:3] ("liq_sustained_ending", "true")
2024/07/11 14:00:23 [show_meta:3] ("liq_true_peak", "1.22")
2024/07/11 14:00:23 [show_meta:3] ("liq_true_peak_db", "1.76 dBFS")
2024/07/11 14:00:23 [show_meta:3] ("media_id", "323883")
2024/07/11 14:00:23 [show_meta:3] ("replaygain_reference_loudness", "-18.00 LUFS")
2024/07/11 14:00:23 [show_meta:3] ("replaygain_track_gain", "-4.78 dB")
2024/07/11 14:00:23 [show_meta:3] Now playing: Nite Radio - Nachrichten
2024/07/11 14:00:23 [decoder.ffmpeg:3] Requested content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}
2024/07/11 14:00:23 [decoder.ffmpeg:3] FFmpeg recognizes "/var/azuracast/stations/niteradio/media/time/news-de.mp3" as audio: {codec: mp3, 48000Hz, 2 channel(s)}
2024/07/11 14:00:23 [decoder.ffmpeg:3] Decoded content-type for "/var/azuracast/stations/niteradio/media/time/news-de.mp3": {audio=pcm(stereo)}

...

2024/07/11 14:00:25 [lang:3] API feedback - Sending POST request to 'http://127.0.0.1:6010/api/internal/1/liquidsoap/feedback' with body: {"replaygain_track_peak":"1.22","liq_fade_out":"2.40","liq_amplify_adjustment":"0.00 dB","song_id":"89f90c2380b5ff2053b95c999f32feba","liq_cue_duration":"164.50","liq_cue_out":"164.50","liq_true_peak":"1.22","liq_fade_out_delay":"0.10","liq_reference_loudness":"-18.00 LUFS","liq_longtail":"false","replaygain_reference_loudness":"-18.00 LUFS","liq_blank_skipped":"false","liq_fade_in_delay":"2.50","artist":"Nite Radio","replaygain_track_gain":"-4.78 dB","replaygain_track_range":"4.45 dB","liq_cross_duration":"2.50","liq_loudness_range":"4.45 LU","title":"Nachrichten","liq_cue_in":"0.00","playlist_id":"539","liq_sustained_ending":"true","liq_autocue":"cue_file","liq_fade_in":"0.10","liq_loudness":"-13.22 LUFS","liq_cue_file":"true","media_id":"323883","liq_cross_start_next":"162.10","liq_amplify":"-4.78 dB","liq_blankskip":"0.00","liq_true_peak_db":"1.76 dBFS"}
2024/07/11 14:00:25 [lang:3] API feedback - Response (200): true
```

Note `liq_cue_file` stays `true` now, and the message is given that cue_file values shall override:
```
2024/07/11 13:59:45 [autocue.cue_file:3] cue_file results override existing metadata because liq_cue_file=true tells us to.
```
